src/couch_replicator/src/couch_replicator_auth_session.erl

```diff
@@ -0,0 +1,486 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_replicator_auth_session).
+
+
+-behavior(couch_replicator_auth).
```

Comment:
also add gen_server behavior?

Reply:
Speaking of gen_servers, does this get added to a supervision tree?

Reply:
Not it doesn't get added, if it crashes, it also crashes the spawning program since it does a start_link. The replication task is also in a way a poor-person's supervisor as it sets the `process_flag(trap_exit, true)` flag.